src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -419,22 +426,29 @@ private void AddResponseCookies(Uri serverUri, HttpResponseMessage response)
             }
         }
 
-		private static NetworkCredential GetCredentials(ICredentials credentials, Uri requestUri)
+        private static Tuple<NetworkCredential, ulong> GetCredentials(ICredentials credentials, Uri requestUri)
         {
-            if (credentials == null)
-            {
-                return null;
-            }
+            NetworkCredential nc = null;
+            ulong curlAuthScheme = CURLAUTH.None;
 
-            foreach (var authScheme in s_authenticationSchemes)
+            if (credentials != null)
             {
-                NetworkCredential networkCredential = credentials.GetCredential(requestUri, authScheme);
-                if (networkCredential != null)
+                for (int i=0; i < s_authSchemePriorityOrder.Length; i++)
                 {
-                    return networkCredential;
+                    NetworkCredential networkCredential = credentials.GetCredential(requestUri, s_authenticationSchemes[i]);
+                    if (networkCredential != null)
+                    {
+                        curlAuthScheme |= s_authSchemePriorityOrder[i];
+                        if (nc == null)
+                        {
+                            nc = networkCredential;
```

Comment:
I'm not quite following the desired outcome here.  We set nc to the first valid NetworkCredential, but we continue to build up curlAuthScheme to include all matches?  What happens if the network credential for one auth scheme was different for the one for another auth scheme?